src/reyna/Connection/ConnectionInfo.cs

```diff
@@ -62,6 +62,76 @@ public bool Mobile
             }
         }
 
+        public bool Wifi
+        {
+            get
+            {
+                try
+                {
+                    var interfaces = NetworkInterface.GetAllNetworkInterfaces();
+                    foreach (var ni in interfaces)
+                    {
+                        if (WifiNetwork(ni))
+                        {
+                            return true;
+                        }
+                    }
+                }
+                catch (Exception)
+                {
+                }
+
+                return false;
+            }
+        }
+
+        public bool Roaming
```

Comment:
I don't think this is a working code.
in the case we don't know if a code will five us the expected result, we do a spike to find the working code then commented it out and TDD it again.

Please read this article to find if the device is roaming or not

We have to read a DWORD value from registry 
#define SN_PHONENOSIM_PATH TEXT("System\\State\\Phone")
#define SN_PHONENOSIM_VALUE TEXT("Status")

#ifndef SN_PHONEROAMING_BITMASK// PhoneRoaming
#define SN_PHONEROAMING_BITMASK 0x200
#endif


then apply a mask 
https://msdn.microsoft.com/en-us/library/aa456240.aspx

@Youhana-Hana